embedded-hal-bus/src/spi.rs

```diff
@@ -1,86 +1,78 @@
 //! SPI bus sharing mechanisms.
 
-/// SPI bus sharing with blocking traits
-pub mod blocking {
-    use core::fmt::Debug;
-    use embedded_hal::{
-        digital::blocking::OutputPin,
-        spi::{
-            blocking::{SpiBusFlush, SpiDevice},
-            Error, ErrorKind, ErrorType,
-        },
-    };
+use core::fmt::Debug;
```

Comment:
I think e-h-bus should keep its `blocking` module so that we can put async sharing facilities here as well. otherwise we will need yet another `embedded-hal-async-bus` crate.
An alternative would be including such stuff in the `e-h-async` crate, but it seems rather unsymmetrical.

Reply:
I think it's cleaner to put all the structs in the same module, because there are some that can impl both blocking+async at the same time. For example, for ExclusiveDevice:

```rust
pub struct ExclusiveDevice<BUS, CS> {..}
impl<BUS, CS> ErrorType for ExclusiveDevice<BUS, CS> where BUS: ErrorType, ... {}
impl<BUS, CS> embedded_hal::spi::SpiDevice for ExclusiveDevice<BUS, CS> where BUS: embedded_hal::spi::SpiBusFlush, ... {}
impl<BUS, CS> embedded_hal_async::spi::SpiDevice for ExclusiveDevice<BUS, CS> where BUS: embedded_hal_async::spi::SpiBusFlush, ... {}
```

(The ExclusiveDevice in `embedded-hal-async` already does that now) 

(see [discussion](https://matrix.to/#/!BHcierreUuwCMxVqOf:matrix.org/$sYELtnHYHSEfmanWah3I-J8iauPfXeSTTUCIKxICm1k?via=matrix.org&via=psion.agg.io&via=mozilla.org) in today's meeting.)